agent/peering_endpoint.go

```diff
@@ -128,6 +133,10 @@ func (s *HTTPHandlers) PeeringInitiate(resp http.ResponseWriter, req *http.Reque
 		return nil, HTTPError{StatusCode: http.StatusBadRequest, Reason: "PeeringToken is required in the payload when initiating a peering."}
 	}
 
+	if err := structs.ValidateMetaTags(args.Meta); err != nil {
```

Comment:
This should actually happen in `agent/rpc/peering/service.go`, as that runs on the (trusted) servers instead of the (less trusted) clients (where the json api happens). Generally you want correctness validation to happen on the servers so you prevent bad data from being recorded.

Reply:
that's a great call out, thank you!

I recall a school of thought that encouraged both:

* do client side validation
* do the same validation on the server side too

what do you think about doing that here as well? Would that make sense form a defense in a depth perspective or just redundant in this case?

Reply:
Chatted offline; we probably don't want to have this validation on the client side too as it would make upgrades tricky. Moved to `agent/rpc/peering/service.go`